### PR TITLE
Update doc to include IS-10

### PR DESCRIPTION
**Pull request by lo-simon**
Update documents and diagrams to include `IS-10` implementation.

### CODE REVIEW

**Comment by lo-simon** [comment]
Thanks, @garethsb, you are right, I must be using the IS-12 branch while doing the document update, I will fix it now.

**Comment by lo-simon** [comment]
> _Edit:_ Apart from the actual change mentioned in the commit comment "highest_pri and lowest_pri are used in Node only", of course!!

When Authorization was implemented, `highest_pri` and `lowest_pri` were also used for the Authorization API, but then during the implementation phase, I noticed the Authorization API will well be not be running on the Registry, i.e. should not be using the same `pri` range as the Registration/Query API.

File: Documents/Architecture.md [hunk 1]
```diff
@@ -118,9 +119,11 @@ for (;;)
 > [nmos/events_api.cpp](../Development/nmos/events_api.cpp),
```
Comment (garethsb) [inline Documents/Architecture.md:119]:
```suggestion
> [nmos/events_api.cpp](../Development/nmos/events_api.cpp),
> [nmos/channelmapping_api.cpp](../Development/nmos/channelmapping_api.cpp),
```

File: Documents/Architecture.md [hunk 2]
```diff
@@ -118,9 +119,11 @@ for (;;)
...
 > [nmos/registration_api.cpp](../Development/nmos/registration_api.cpp),
 > [nmos/query_api.cpp](../Development/nmos/query_api.cpp),
-> [nmos/system_api.cpp](../Development/nmos/system_api.cpp)
+> [nmos/system_api.cpp](../Development/nmos/system_api.cpp),
+> [nmos/channelmapping_api.cpp](../Development/nmos/channelmapping_api.cpp)
+> [nmos/authorization_redirect_api.cpp](../Development/nmos/authorization_redirect_api.cpp)
```
Comment (garethsb) [inline Documents/Architecture.md:124]:
```suggestion
> [nmos/system_api.cpp](../Development/nmos/system_api.cpp),
> [nmos/authorization_redirect_api.cpp](../Development/nmos/authorization_redirect_api.cpp)
```

File: Documents/Dependencies.md [hunk 3]
```diff
@@ -325,4 +325,4 @@ No installation is necessary.
 
 # What Next?
 
-These [instructions](Getting-Started.md) explain how to build nmos-cpp itself, run the test suite, and try out the registry and node applications.
+These [instructions](Getting-Started.md) explain how to build nmos-cpp itself, run the test suite, and try out the registry and node applications.
```
Comment (garethsb) [inline Documents/Dependencies.md:328]: What's changed here? 🤔

File: Documents/Architecture.md [hunk 4]
```diff
@@ -1,10 +1,10 @@
 # Architecture of nmos-cpp
 
-The [nmos](../Development/nmos/) module fundamentally provides three things. 
+The [nmos](../Development/nmos/) module fundamentally provides three things.
 
-1. A C++ data model for the AMWA IS-04 and IS-05 NMOS resources which represent the logical functionality of a Node, or equally, for the resources of many Nodes held by a Registry.
-2. An implementation of each of the REST APIs defined by the AMWA IS-04 and IS-05 NMOS specifications, in terms of the data model.
-3. An implementation of the Node and Registry "active behaviours" defined by the specifications.  
+1. A C++ data model for the AMWA IS-04, IS-05 and IS-08 NMOS resources which represent the logical functionality of a Node, or equally, for the resources of many Nodes held by a Registry.
+2. An implementation of each of the REST APIs defined by the AMWA IS-04, IS-05, IS-08, IS-09 and IS-10 NMOS specifications, in terms of the data model.
```
Comment (garethsb) [inline Documents/Architecture.md:6]: Suggest simplifying here, so it's one less place to keep up to date as we implement additional specs?

```suggestion
1. A C++ data model for the AMWA NMOS resources which represent the logical functionality of a Node, or equally, for the resources of many Nodes held by a Registry.
2. An implementation of each of the REST APIs defined by the AMWA NMOS specifications, in terms of the data model.
```

File: Documents/images/Authorization-Client-Registration.png
Comment (jonathan-r-thorpe) [inline Documents/images/Authorization-Client-Registration.png:1]: This diagram still shows the Device Code Flow grant type in it which should perhaps be removed?

File: Documents/images/Authorization-Code-Flow.png
Comment (jonathan-r-thorpe) [inline Documents/images/Authorization-Code-Flow.png:1]: Similar to the Client Credentials diagram, the Authorization Process box at the bottom of the diagram contains an Authorization Code Flow box suggesting an infinite recursion :|

File: Documents/images/Authorization-behaviour.png
Comment (jonathan-r-thorpe) [inline Documents/images/Authorization-behaviour.png:1]: Not sure what the "Is required to access protected API?" diamond is asking.  Also on "No" it enters the "do authorization_operation" loop which only seems to exit on Failed - is that right?

Comment (jonathan-r-thorpe) [reply Documents/images/Authorization-behaviour.png:1]: If it's a separate process (like the "start authorization code flow" box) should it perhaps be "start authorization_operation loop"? (same applies I guess to the authorization_operation box at the bottom of the diagram)

File: Documents/images/Authorization-Client-Registration.png
Comment (lo-simon) [reply Documents/images/Authorization-Client-Registration.png:1]: Well spotted, I will remove `Device Code Flow grant`

File: Documents/images/Authorization-Client-Credentials-Flow.png
[the remark this answers is not in the document]
Comment (lo-simon) [reply Documents/images/Authorization-Client-Credentials-Flow.png:1]: I am trying to show once the connection is broken between Auth server & the Node, Node will start over again to fetch Auth server metadata.... and Client Credentials flow

**Pull request merged**